### ISSUE TITLE
Regression does not work with Python APIs

### ISSUE THREAD

**Issue by slaweks17** [Microsoft/CNTK#1094]
Hi,
I have been trying unsuccessfully to run a simple vector-to-vector regression using Python APIs (beta 4 and 5).
It appears to me that there is something wrong with squared_error, because my input and output vectors are roughly between -1 and 1, batch size is 25, but squared error reported is 38565287677857290518528.000000 * 3200 and does not change during training almost at all.
I attach the code (which is based on your FeedForwardNet.py example). BTW, original example works OK, also on my GPU.

Regards,
Slawek

P.S. 
The code below does not look good, so I also attach the file.

#-------------------
import numpy as np
import sys
import os
from cntk.device import gpu, set_default_device
from cntk import Trainer
from cntk.learner import sgd, learning_rate_schedule, UnitType
from cntk.ops import *
from cntk.utils import ProgressPrinter
from cntk.initializer import glorot_uniform, he_normal

def linear_layer(input_var, output_dim):
    times_param = parameter(shape=(list(input_var.shape)+[output_dim]), init=glorot_uniform())
    bias_param = parameter(shape=(output_dim), init=0)

    t = times(input_var, times_param)
    return bias_param + t

def fully_connected_layer(input, output_dim, nonlinearity):
    p = linear_layer(input, output_dim)
    return nonlinearity(p)

def fully_connected_net(input, output_dim, hidden_layer_dim, num_hidden_layers, nonlinearity):
    r = fully_connected_layer(input, hidden_layer_dim, nonlinearity)
    for i in range(1, num_hidden_layers):
        r = fully_connected_layer(r, hidden_layer_dim, nonlinearity)
    return linear_layer(r, output_dim)

def generate_random_data(minibatch_size, input_dim, output_dim):
    X = np.random.randn(minibatch_size, input_dim)*0.5
    X = X.astype(np.float32)
    Y=X+0.5
    return X, Y

set_default_device(gpu(0))

input_dim = 2
output_dim = 2
num_hidden_layers = 2
hidden_layers_dim = 50

input = input_variable((input_dim), np.float32)
label = input_variable((output_dim), np.float32)

netout = fully_connected_net(
    input, output_dim, hidden_layers_dim, num_hidden_layers, tanh)

ce = squared_error(netout, label)
pe = squared_error(netout, label)

lr_per_minibatch=learning_rate_schedule(0.5, UnitType.minibatch)
trainer = Trainer(netout, ce, pe, sgd(netout.parameters, lr=lr_per_minibatch))
minibatch_size = 25

pp = ProgressPrinter(128)
for i in range(1024):
    features, labels = generate_random_data(
        minibatch_size, input_dim, output_dim)
    trainer.train_minibatch({input: features, label: labels})
    pp.update_with_trainer(trainer)
    
pp.epoch_summary()
test_features, test_labels = generate_random_data(
    minibatch_size, input_dim, output_dim)
avg_error = trainer.test_minibatch(
    {input: test_features, label: test_labels})
print(" error rate on an unseen minibatch %f" % avg_error)

**Comment by slaweks17** [comment]
Hi,
Update: Yes, I think there is something wrong with the squared_error, because when I replaced it with my own loss function
def PinBallLoss (z, t, tau,  weight=1):
  labelDim=t.shape   
  tauV=Constant(tau, shape=labelDim, dtype=np.float32)
  tau1mV=Constant(1-tau, shape=labelDim, dtype=np.float32)
  loss=weight*reduce_sum(element_select( greater_equal(t, z), element_times(minus(t,z), tauV), element_times(minus(z,t), tau1mV))) 
  return loss

the regression seems to be working.
BTW, could someone please comment if this loss function is efficiently coded?

Regards,
Slawek

**Comment by cha-zhang** [comment]
In your first post, if you change your learning rate from 0.5 to 0.1, you get: 

```
(python34) D:\temp\cntk>python test.py
 Minibatch[   1- 128]: loss = 0.018406 * 3200
 Minibatch[ 129- 256]: loss = 0.001144 * 3200
 Minibatch[ 257- 384]: loss = 0.000846 * 3200
 Minibatch[ 385- 512]: loss = 0.000733 * 3200
 Minibatch[ 513- 640]: loss = 0.000659 * 3200
 Minibatch[ 641- 768]: loss = 0.000753 * 3200
 Minibatch[ 769- 896]: loss = 0.000591 * 3200
 Minibatch[ 897-1024]: loss = 0.000694 * 3200
Finished Epoch [1]: loss = 0.002978 * 25600 2.235s (11452.8 samples per second)
 error rate on an unseen minibatch 0.003646
```
So I think this is probably not a SquareError issue?

**Comment by wolfma61** [comment]
No responce on this one to Cha's question ... the referenced issue #1160 is closed.
I'm closing the issue ...
If this still occurs with the most recent released version, please reopen (or create a new issue) with sufficient information to reproduce the problem
